Nextflow installation issues in GHA

Hey!,

i am running into various issues with the Nextflow installation in my Github action runs:
 a couple of examples:

 - https://github.com/nf-core/sarek/actions/runs/3456705070/jobs/5769631081:
```
3s
Run nf-core/setup-nextflow@v1
Error: Could not retrieve Nextflow release matching 21.10.3.
API rate limit exceeded for installation ID 1565271.
Error: Could not parse the download URL
Cannot read properties of undefined (reading 'filter')
Error: versionSpec parameter is required
Warning: Nextflow appears to have installed correctly, but an error was thrown while running it.
```
- or this one:
https://github.com/nf-core/modules/actions/runs/3454606071/jobs/5766033679

```
Run wget -qO- get.nextflow.io | bash
Downloading nextflow dependencies. It may require a few seconds, please wait .. 
Downloading nextflow dependencies. It may require a few seconds, please wait .. 
Unable to initialize nextflow environment

Error: Process completed with exit code 1.
```

--- a/src/functions.ts
+++ b/src/functions.ts
@@ -111,3 +111,22 @@ export async function install_nextflow(
 
   return temp_install_dir
 }
+
+export function check_cache(version: string): boolean {
+  const cleaned_version = semver.clean(version, true)
+  if (cleaned_version === null) {
+    return false
+  }
+  const resolved_version = String(cleaned_version)
+
+  const nf_path = tc.find("nextflow", resolved_version)
+  if (!nf_path) {
+    core.debug(`Could not find Nextflow ${resolved_version} in the tool cache`)
+    return false
+  } else {
+    core.debug(`Found Nextflow ${resolved_version} at path '${nf_path}'`)
+    core.debug(`Adding '${nf_path}' to PATH`)
+    core.addPath(nf_path)
+    return true
+  }
+}

--- a/src/main.ts
+++ b/src/main.ts
@@ -4,8 +4,14 @@ import * as github from "@actions/github"
 import { GitHub } from "@actions/github/lib/utils"
 import * as tc from "@actions/tool-cache"
 import * as fs from "fs"
+import semver from "semver"
 
-import { install_nextflow, nextflow_bin_url, release_data } from "./functions"
+import {
+  check_cache,
+  install_nextflow,
+  nextflow_bin_url,
+  release_data
+} from "./functions"
 
 async function run(): Promise<void> {
   // Set environment variables
@@ -16,7 +22,10 @@ async function run(): Promise<void> {
   const version = core.getInput("version")
   const get_all = core.getBooleanInput("all")
 
-  let resolved_version = ""
+  // Check the cache for the Nextflow version that matched last time
+  if (check_cache(version)) {
+    return
+  }
 
   // Setup the API
   let octokit: InstanceType<typeof GitHub> | undefined
@@ -32,6 +41,7 @@ async function run(): Promise<void> {
 
   // Get the release info for the desired release
   let release = {}
+  let resolved_version = ""
   try {
     if (octokit !== undefined) {
       release = await release_data(version, octokit)
@@ -60,24 +70,19 @@ async function run(): Promise<void> {
   }
   try {
     // Download Nextflow and add it to path
-    let nf_path = ""
-    nf_path = tc.find("nextflow", resolved_version)
-
-    if (!nf_path) {
-      core.debug(`Could not find Nextflow ${resolved_version} in cache`)
+    if (!check_cache(resolved_version)) {
       const nf_install_path = await install_nextflow(url, resolved_version)
-
-      nf_path = await tc.cacheDir(nf_install_path, "nextflow", resolved_version)
+      const cleaned_version = String(semver.clean(resolved_version, true))
+      const nf_path = await tc.cacheDir(
+        nf_install_path,
+        "nextflow",
+        cleaned_version
+      )
+      core.addPath(nf_path)
+      core.info(`Downloaded \`nextflow\` to ${nf_path} and added to PATH`)
       core.debug(`Added Nextflow to cache: ${nf_path}`)
-
       fs.rmdirSync(nf_install_path, { recursive: true })
-    } else {
-      core.debug(`Using cached version of Nextflow: ${nf_path}`)
     }
-
-    core.addPath(nf_path)
-
-    core.info(`Downloaded \`nextflow\` to ${nf_path} and added to PATH`)
   } catch (e: unknown) {
     if (e instanceof Error) {
       core.setFailed(e.message)

